staging/src/k8s.io/apimachinery/pkg/api/validation/objectmeta.go

```diff
@@ -261,6 +289,28 @@ func ValidateObjectMetaAccessorUpdate(newMeta, oldMeta metav1.Object, fldPath *f
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("deletionTimestamp"), newMeta.GetDeletionTimestamp(), "field is immutable; may only be changed via deletion"))
 	}
 
+	oldInit := oldMeta.GetInitializers()
+	newInit := newMeta.GetInitializers()
+	switch {
+	case oldInit == nil && newInit != nil:
+		// Initializers may not be set on new objects
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("initializers"), nil, "field is immutable once all initialization has completed"))
+	case oldInit != nil && newInit == nil:
+		// this is a valid transition and means initialization was successful
+	case oldInit != nil && newInit != nil:
+		// validate changes to initializers
+		switch {
+		case oldInit.Result == nil && newInit.Result != nil:
+			// setting a result is allowed
+			allErrs = append(allErrs, validateInitializersResult(newInit.Result, fldPath.Child("initializers", "result"))...)
+		case oldInit.Result != nil:
```

Comment:
Right now, `Result` == `failed`.  A failed initialization is final, and no updates can be executed after this point.  We can always relax that in the future.

Why is comment not correct?  We don't allow changing result once set - I can add the "once set" bit.